woocommerce/woocommerce
# [Payments NOX] e2e drag and drop functionality for providers and offline PMs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #53247 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm i && pnpm build`
2. Go to the payment settings page
3. Verify that you can see the drag handles at the left of each payment provider in the list.
4. Open the network tab in the browser console, perform a drag and drop - verify you see a `/order` request in the network tab - verify the request payload looks sensible - the `order_map` should be posted, and it should contain a list of payment providers - the ordering indexes should be swapped, so for example if PayPal has ordering ID 20, and WooPayments has ordering ID 10, and you swap them, the order map request should look like:

```json
{
    "woopayments": 20,
    "paypal": 10,
}
```



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
